sda-auth/main.go

```diff
@@ -390,6 +393,16 @@ func main() {
 	app.Get("/elixir/login", authHandler.getElixirLogin)
 	app.Get("/elixir/cors_login", authHandler.getElixirCORSLogin)
 
+	publicKey, err := readPublicKeyFile(authHandler.Config.PublicFile)
+	if err != nil {
+		log.Info("Failure to get public key: ", err)
+	} else {
+		authHandler.pubKey = hex.EncodeToString(publicKey[:])
+	}
```

Comment:
```suggestion
	publicKey, err := readPublicKeyFile(Config.PublicFile)
	if err != nil {
		log.Fatal("Failure to get public key: ", err)
	}
```

Move this to line 350, then you can initialize the `authHandler` struct directly with `pubkey:	hex.EncodeToString(publicKey[:])`

Reply:
Right, I could do these changes, it would look a bit tidier.
the log should not be Fatal though, right? 
You argued about providing the /info endpoint conditionally if the public key exists.
I thought that was about a scenario where it's missing?

Reply:
> You argued about providing the /info endpoint conditionally if the public key exists.
I thought that was about a scenario where it's missing from the deployment?

That will not be the case anymore since the public key is part of the mandatory configuration options now.

Reply:
Yes, that is fine. Since the public key is mandatory we should fail hard if that file can not be read or is not a public key.